serializers.go

```diff
@@ -83,7 +89,28 @@ type AvroJSONSerializer struct {
 }
 
 func (s *AvroJSONSerializer) Marshal(metric map[string]interface{}) ([]byte, error) {
-	return s.codec.TextualFromNative(nil, metric)
+	binaryValue, err := s.codec.BinaryFromNative(nil, metric)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// if using schema registry encode a header with schema ID
+	if schemaRegistryId != 0 {
+		var header []byte
+
+		// first byte is magic byte, always 0 for now
+		header = append(header, byte(0))
+
+		//4-byte schema ID as returned by the Schema Registry
+		binarySchemaId := make([]byte, 4)
+		binary.BigEndian.PutUint32(binarySchemaId, uint32(schemaRegistryId))
```

Comment:
Every a message is marshal we produce a little bit of garbage. Can we generate `binarySchemaId` once when we discover the `schemaRegistryId` to avoid the allocation here on every message?